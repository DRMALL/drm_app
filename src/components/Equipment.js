import React, { Component } from 'react'
import { View, Text, Image, ListView, ScrollView, TouchableOpacity } from 'react-native'
import { equipmentName } from '../common/strings'
import { subTitleColor, primaryColor } from '../common/constants'
import { equipment } from '../styles'

const gobackWhiteIcon = require('../images/navigation_icons/goback_white.png')
const emptyIcon = require('../images/navigation_icons/empty.png')
const pic4 = require('../images/pic4.png')
const dropdownNormal = require('../images/dropdown_normal.png')

export default class Equipment extends Component {
  static navigationOptions = ({ navigation })=> ({
    headerStyle: {
      backgroundColor: primaryColor,
    },
    headerTitle: <Text style={{ fontSize: 20, color: '#FFF', alignSelf: 'center' }} >{equipmentName}</Text>,
    headerLeft: <TouchableOpacity style={{paddingLeft: 20}} onPress={() => navigation.goBack()}>
      <Image source={gobackWhiteIcon}/>
    </TouchableOpacity>,
    headerRight: <Image style={{marginLeft: 20}} source={emptyIcon}/>,
  })
  render() {
    return (
      <ScrollView style={equipment.wrap}>
        <Text style={equipment.fixText}>运行状态</Text>
        <Text style={equipment.stateText}>正常</Text>
        <Text style={equipment.fixText}>运行图示（点击图片放大）</Text>
        <View style={equipment.imgView}>
          <Image style={equipment.img} source={pic4}/>
        </View>
        <View style={equipment.twoTextView}>
          <Text style={equipment.fix2Text}>指标数据</Text>
          <Text style={[equipment.fix2Text, {position: 'absolute', right: 15}]}>更新时间: 2017-07-19 19:30</Text>
        </View>
        <View style={equipment.dataView}>
          <TouchableOpacity style={equipment.dataTouch} onPress={()=> 'touch'}>
            <Text style={equipment.textTouch}>本类别数据名称</Text>
            <Image style={equipment.imgTouch} source={dropdownNormal}/>
          </TouchableOpacity>
          <TouchableOpacity style={equipment.dataTouch} onPress={()=> 'touch'}>
            <Text style={equipment.textTouch}>本类别数据名称</Text>
            <Image style={equipment.imgTouch} source={dropdownNormal}/>
          </TouchableOpacity>
          <TouchableOpacity style={equipment.dataTouch} onPress={()=> 'touch'}>
            <Text style={equipment.textTouch}>本类别数据名称</Text>
            <Image style={equipment.imgTouch} source={dropdownNormal}/>
          </TouchableOpacity>
          <TouchableOpacity style={equipment.dataTouch} onPress={()=> 'touch'}>
            <Text style={equipment.textTouch}>本类别数据名称</Text>
            <Image style={equipment.imgTouch} source={dropdownNormal}/>
          </TouchableOpacity>
        </View>
        <Text style={equipment.fixText}>运行日志</Text>
        <View style={equipment.logView}>
          <Text style={equipment.logText}>2017-05-05 15:21:32  设备点火</Text>
          <Text style={equipment.logText}>2017-05-05 15:21:32  设备点火</Text>
          <Text style={equipment.logText}>2017-05-05 15:21:32  设备点火</Text>
          <Text style={equipment.logText}>2017-05-05 15:21:32  设备点火</Text>
          <Text style={equipment.logText}>2017-05-05 15:21:32  设备点火</Text>
        </View>
      </ScrollView>
    )
  }
}