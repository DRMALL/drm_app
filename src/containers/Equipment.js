import React, { Component } from 'react'
import { View, Text, Image, ListView, ScrollView, TouchableOpacity, Alert } from 'react-native'
import moment from 'moment'
import socket from 'socket.io-client'
import { Navigator } from 'react-native-deprecated-custom-components'
import Lightbox from 'react-native-lightbox'
import { equipmentName, equipmentRunState, equipmentRunImg, equipmentIndexData, indexDataUpdateTime, equipmentRunningLog, internalServerError, tokenKey } from '../common/strings'
import { subTitleColor, primaryColor } from '../common/constants'
import { equipment } from '../styles'
import { equipmentDataList, equipmentLogList } from '../utils/virtualData'
import IndexData from '../components/equipment/IndexData'
import RunningLog from '../components/equipment/RunningLog'
import { checkToken } from '../utils/handleToken'
import { getPort } from '../utils/fetchMethod'
import { getMoniterdevsNum } from '../apis'

import store from '../utils/store'
import statuAC from '../actions/statuAC'

const gobackWhiteIcon = require('../images/navigation_icons/goback_white.png')
const emptyIcon = require('../images/navigation_icons/empty.png')
const pic4 = require('../images/pic4.png')

export default class Equipment extends Component {
  static navigationOptions = ({ navigation })=> ({
    headerStyle: {
      backgroundColor: primaryColor,
    },
    headerTitle: <Text style={{ fontSize: 20, color: '#FFF', alignSelf: 'center' }} >{navigation.state.params.statuItemNumber || '00000000'}</Text>,
    headerLeft: <TouchableOpacity style={{padding: 10, paddingLeft: 20}} onPress={() => navigation.goBack()}>
      <Image source={gobackWhiteIcon}/>
    </TouchableOpacity>,
    headerRight: <Image style={{marginLeft: 20}} source={emptyIcon}/>,
  });

  constructor(props) {
    super(props)
    this.state = store.getState().statu
  }

  componentDidMount() {
    statuAC.getEquipData({})
    this.getMonNumData()
  }

  componentWillMount() {
    this.unsubscribe = store.subscribe(()=> this.setState(store.getState().statu) )
  }

  componentWillUnmount(){
    this.unsubscribe()
  }

  getMonNumData() {    //获取数据库数据，存eqNumberData
    checkToken(tokenKey)
    .then(async token => {
      let { statuItemNumber } = this.props.navigation.state.params
      let res = await getPort(`${getMoniterdevsNum}?number=${statuItemNumber}&token=${token}`)
      if(!res) {
        Alert.alert('错误', internalServerError,
          [ {text: 'OK', onPress: () => 'OK'}, ],
          { cancelable: false }
        )
      } else if(res.code == 200) {
        // if(res.data === null) {
        //   Alert.alert('提示', '暂无内容,请刷新再试',
        //     [ {text: 'OK', onPress: () => 'ok'}, ],
        //     { cancelable: false }
        //   )
        //   this.props.navigation.goBack()
        // } else 
        statuAC.setEqNumItemData(res.data)
      } else {
        Alert.alert('错误', JSON.stringify(res.message),
          [ {text: 'OK', onPress: () => 'OK'}, ],
          { cancelable: false }
        )
      }
    })
  }

  render() {
    let { equipmentData, eqNumberData } = this.state
      , { statuItemNumber } = this.props.navigation.state.params
      , isNormal = false
      , equipmentItemData = {}
    equipmentData.map((eqItem, index)=> {
      if(statuItemNumber == eqItem.number) {
        isNormal = true
        equipmentItemData = eqItem
      }
    })
    return (
      <ScrollView style={equipment.wrap}>
        <Text style={equipment.fixText}>{equipmentRunState}</Text>
        <Text style={equipment.stateText}>{isNormal ? '正常运行' : '离线'}</Text>
        
        <View style={equipment.twoTextView}>
          <Text style={equipment.fix2Text}>{equipmentIndexData}</Text>
          <Text style={[equipment.fix3Text, {position: 'absolute', right: 15}]}>{
            indexDataUpdateTime + `${equipmentItemData.rnTimestamp ? moment(equipmentItemData.rnTimestamp).format('YYYY-MM-DD HH:mm') : (
              eqNumberData ? moment(eqNumberData.updatedAt).format('YYYY-MM-DD HH:mm') : '0000-00-00 00:00'
            )}`
          }</Text>
        </View>
        <View style={equipment.dataView}>
          <IndexData indexData={equipmentDataList} equipmentItemData={equipmentItemData} eqNumberData={eqNumberData} {...this.props}/>
        </View>
        
      </ScrollView>
    )
  }
}

// {
//   <Text style={equipment.fixText}>{equipmentRunImg}</Text>
//         <Lightbox style={equipment.imgView}>
//           <Image style={equipment.img} source={pic4}/> 
//         </Lightbox>
//   <Text style={equipment.fixText}>{equipmentRunningLog}</Text>
//         <View style={equipment.logView}>
//           { equipmentLogList.map((log, i)=> <RunningLog key={i} log={log} /> ) }
//         </View>
// }