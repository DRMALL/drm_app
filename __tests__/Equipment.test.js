import 'react-native'
import React from 'react'
import Equipment from '../src/containers/Equipment'
import renderer from 'react-test-renderer';

test('Equipment test', () => {
  const navigation = {
    state: {
      params: {
        statuItemData: false,
      },
    },
  }
  const tree = renderer.create(
    <Equipment navigation={ navigation }/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})