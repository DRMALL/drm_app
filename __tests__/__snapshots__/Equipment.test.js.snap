// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Equipment test 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#F7F7F7",
      "width": "100%",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "borderColor": "rgba(0,0,0,.06)",
          "borderWidth": 0.5,
          "color": "rgba(0,0,0,.3)",
          "fontSize": 16,
          "height": 40,
          "paddingLeft": 15,
          "paddingVertical": 9,
        }
      }
    >
      运行状态
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "#fff",
          "fontSize": 18,
          "height": 60,
          "paddingLeft": 15,
          "paddingVertical": 19,
        }
      }
    >
      正常
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "rgba(0,0,0,.06)",
          "borderWidth": 0.5,
          "flexDirection": "row",
          "height": 40,
          "paddingLeft": 15,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0,0,0,.3)",
            "fontSize": 16,
          }
        }
      >
        指标数据
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,.6)",
              "fontSize": 12,
            },
            Object {
              "position": "absolute",
              "right": 15,
            },
          ]
        }
      >
        更新时间: 2017-09-04 09:13
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.06)",
          }
        }
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "rgba(0,0,0,.06)",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 60,
                "opacity": 1,
                "paddingLeft": 15,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "color": "rgba(0,0,0,.6)",
                  },
                ]
              }
            >
              本类别数据名称1
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "position": "absolute",
                  "right": 15,
                  "width": 15,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
              }
            }
          />
        </View>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "rgba(0,0,0,.06)",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 60,
                "opacity": 1,
                "paddingLeft": 15,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "color": "rgba(0,0,0,.6)",
                  },
                ]
              }
            >
              本类别数据名称2
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "position": "absolute",
                  "right": 15,
                  "width": 15,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
              }
            }
          />
        </View>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "rgba(0,0,0,.06)",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 60,
                "opacity": 1,
                "paddingLeft": 15,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "color": "rgba(0,0,0,.6)",
                  },
                ]
              }
            >
              本类别数据名称3
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "position": "absolute",
                  "right": 15,
                  "width": 15,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
              }
            }
          />
        </View>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "rgba(0,0,0,.06)",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 60,
                "opacity": 1,
                "paddingLeft": 15,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "color": "rgba(0,0,0,.6)",
                  },
                ]
              }
            >
              本类别数据名称4
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "position": "absolute",
                  "right": 15,
                  "width": 15,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
